guides/updating/update-indy-sdk-to-askar.md

```diff
@@ -0,0 +1,168 @@
+
+The Indy-sdk and Aries Askar have different database structures. So first of
+all we need to change some table names, decrypt all the items with the old Indy
+keys, encrypt the items with the new Aries Askar keys and store them inside the
+new structure.
+
+### Try to open the wallet in the new Aries Askar structure
+
+When the wallet is correctly transformed, the wallet will be attempted to be
+openend.
+
+### Update the keys
+
+Aries Askar has a specific way to store keys and every key, defined by the
+category of `Indy::Key` will be migrated.
+
+### Update the credential definitions
+
+:::danger
+
+Migration of credential definitions is not yet supported. This is why it is
+strongly advised not to run this script in a node.js environment.
+
+:::
+
+### Update the link secret(s) (master secret)
+
+The link sercets, identified by the category `Indy::MasterSecret`, are updated
+next. They are stored inside a new `AnonCredsLinkSecretRecord`.
+
+:::caution
+
+Since we have to set a default link secret, some additional logic is added to
+detect this. You can either supply a `defaultLinkSercetId` as a constructor
+parameter or it will be based on your `walletId`.
+
+If there is no Indy record with the `defaultLinkSecretId` or the `walletId`, an
+error will be thrown and the migration will be restored.
+
+:::
+
+### Update the credentials
+
+The credentials, identified by the category `Indy::Credential` are updated
+last. They are stored in the new `AnonCredsCredentialRecord` as a one-to-one
+copy. This means that they now support more tags and will make querying a lot
+easier.
+
+### All the other records
+
+All the other records will be transferred without any updates as they are not
+Indy specific.
+
+## How to update
+
+Updating does not require a lot of code, but must be done with caution.
+
+### add the required dependencies:
+
+```sh
+yarn add @hyperledger/aries-askar-react-native @aries-framework/indy-sdk-to-askar-migration react-native-file-system
+```
+
+Below is the minimal code required for the migration to work. It is recommended
+to turn the logger on as it gives a lot of information regarding the migration.
+
+:::caution
+
+The agent is not allowed to be initialized to run this script. This makes sure
+nothing else happens during the migration.
+
+:::
+
+```typescript
+import { IndySdkToAskarMigrationUpdater } from '@hyperledger/indy-sdk-to-askar-migration'
+import { agentDependencies } from '@aries-framework/react-native'
+import { AskarModule } from '@aries-framework/askar'
+
+const oldAgent = new Agent({
+  config: {
+    /* ... */
+  },
+  modules: {
+    modules: { askar: new AskarModule() },
+  },
+  dependencies: agentDependencies,
+})
+
+const dbPath = '' // see section below
+
+const updater = await IndySdkToAskarMigrationUpdater.initialize({ dbPath, agent })
+await updater.update()
+```
+
+### Getting the database path
+
+#### Android
+
+On android, the database is located under the `ExternalDirectoryPath`.
```

Comment:
usually. It can be overrideen using the external storage env variable (see docs in indy-sdk-react-native)

Reply:
Yes, same for iOS. I will add an info message above.